src/components/select/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import './style.less'
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import { getMovies } from '../../services/movieService';
+import { MovieDay, Movie, CinemaMovieSchedule } from '../../models/movieDay';
+import { isSameDay } from '../../services/calendarService';
+import MovieList from '../movie-list';
+
+
+interface State {
+    cities: Array<string>;
+    selectedCity: string;
+    selectedDate: Date;
+    showMovies: boolean;
+    movies: Array<Movie>;
+}
+
+interface Props { }
+
+
+class Select extends React.Component<Props, State> {
+    formattedDate: string;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            cities: ['Vilnius', 'Kaunas', 'Klaipėda', 'Šiauliai', 'Panevėžys'],
+            selectedCity: 'Vilnius',
+            selectedDate: new Date(),
+            showMovies: false,
+            movies: [],
+        };
+
+        this.searchForMovies = this.searchForMovies.bind(this);
+    }
+
+    componentDidMount() {
+        this.formatDate(this.state.selectedDate)
+    }
+
+    showCitiesToSelect() {
+        let cityOptions = [];
+        for (let i = 0; i < this.state.cities.length; i++) {
+            cityOptions.push(<option key={i}>{this.state.cities[i]}</option>)
+        }
+        return cityOptions;
+    }
+
+    handleDateChange = (date: Date): void => {
+        this.setState({
+            selectedDate: date,
+            showMovies: false
+        });
+        this.formatDate(date);
+    };
+
+    selectCity(event: string): void {
+        this.setState({
+            selectedCity: event,
+            showMovies: false,
+        });
+    }
+
+    formatDate(date: Date): string {
+        let options = { month: 'long', day: 'numeric' }
+        this.formattedDate = date.toLocaleDateString('en-US', options);
+        return this.formattedDate;
```

Comment:
Now, return value is never used. If the approach, described in the comment below is used, then the formatDate function would work great, and the field this.formattedDate can be safely deleted!